src/formatDate.js

```diff
@@ -9,6 +9,64 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const oldDateSeparator = fromFormat[3];
+  const splitFromFormat = date.split(oldDateSeparator);
+  const newFormatDate = [];
+  const year = [];
+  let month = '';
+  let day = '';
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    switch (fromFormat[i][0]) {
+      case 'Y':
+        year.push(splitFromFormat[i]);
+        break;
+
+      case 'M':
+        month = splitFromFormat[i];
+        break;
+
+      case 'D':
+        day = splitFromFormat[i];
+        break;
+
+      default:
+        break;
+    }
+  }
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    switch (toFormat[i][0]) {
+      case 'Y':
+        const yearNewFormatLength = toFormat[i].length;
+
+        if (year[0].length === yearNewFormatLength) {
+          newFormatDate.push(...year);
+        } else if (year[0].length > yearNewFormatLength) {
+          newFormatDate.push(year[0].slice(2));
+        } else if (year[0].length < yearNewFormatLength && year[0] < 30) {
```

Comment:
When checking if the year needs to be converted from 2 digits to 4, you are only comparing the string to the number 30. This will not work correctly because the comparison is between a string and a number. You should convert the string to a number before comparing it.